src/arc/mp/MultiplayerSingleton.as

```diff
@@ -511,7 +511,9 @@ package arc.mp
 
         public function gameplaySubmit(room:Object):void
         {
-            if (currentSong.engine != null)
+            var matchSong:Object = currentSong || room.match.song;
+
+            if (matchSong != null && matchSong.engine != null)
```

Comment:
There's a possibility that `matchSong` is `null`, which would fail this check and cause a crash in the code where the game stats string is built.
Perhaps consider changing this check to `if (matchSong == null || matchSong.engine != null)`?